### PR TITLE
Issue 18553

### PR DESCRIPTION
**Pull request by erickgonzalez**
Add Growl Notifications to PP. We have 2 types of Notifications:

* Succes = Lasts 5 seconds.
* Failure = Until the user dismisses it and includes a link to the `Publishing Queue Portlet`. This notification will pop up until the number of tries is reached (3 by default).

If the bundle has a name set it will show it, If not it will show the title and the type of the first items that are in the bundle.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java [hunk 1]
```diff
@@ -355,6 +381,46 @@ private void updateBundleStatus(final PublishAuditStatus bundleAudit,
 			}
 		}
 		Logger.info(this, "===========================================================");
+
+		//Growl Notifications
+		if(UtilMethods.isSet(notificationMessage)){
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:386]: I would extract to a private method `sendNotification`

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java [hunk 2]
```diff
@@ -355,6 +381,46 @@ private void updateBundleStatus(final PublishAuditStatus bundleAudit,
...
+	 */
+	private String generateBundleTitle(final Map<String,String> bundleAssets)
+			throws LanguageException {
+		int count = 0;
+		String bundleTitle = "";
+		for(final String id : bundleAssets.keySet()){
+			if(count < 3) {
+				final String assetType = bundleAssets.get(id);
+				final String assetTitle = PublishAuditUtil.getInstance().getTitle(assetType, id);
+				bundleTitle += "<strong>" + assetType + ":</strong> " + assetTitle + "<br/>";
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:416]: Codacy found an issue: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5889561)

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java [hunk 3]
```diff
@@ -355,6 +381,46 @@ private void updateBundleStatus(final PublishAuditStatus bundleAudit,
...
+	 *
+	 * @param bundleAssets assets that were added manually to the bundle
+	 * @return a string that will be the argument for the notification
+	 */
+	private String generateBundleTitle(final Map<String,String> bundleAssets)
+			throws LanguageException {
+		int count = 0;
+		String bundleTitle = "";
+		for(final String id : bundleAssets.keySet()){
+			if(count < 3) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:413]: Codacy found an issue: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5889561)

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java [hunk 4]
```diff
@@ -355,6 +381,46 @@ private void updateBundleStatus(final PublishAuditStatus bundleAudit,
...
+				final String assetType = bundleAssets.get(id);
+				final String assetTitle = PublishAuditUtil.getInstance().getTitle(assetType, id);
+				bundleTitle += "<strong>" + assetType + ":</strong> " + assetTitle + "<br/>";
+				count++;
+			} else {
+				bundleTitle += "..." + (bundleAssets.keySet().size()-3) + " " + LanguageUtil.get("publisher_audit_more_assets");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:419]: Codacy found an issue: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5889561)

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java [hunk 5]
```diff
@@ -2,10 +2,16 @@
 
 import static com.dotcms.util.CollectionsUtils.map;
 
+import com.dotcms.api.system.event.message.MessageSeverity;
+import com.dotcms.api.system.event.message.MessageType;
+import com.dotcms.api.system.event.message.SystemMessageEventUtil;
+import com.dotcms.api.system.event.message.builder.SystemMessageBuilder;
 import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.enterprise.LicenseUtil;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:10]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.enterprise.LicenseUtil'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5889561)

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java [hunk 6]
```diff
@@ -292,10 +303,19 @@ private void updateAuditStatus(final List<Map<String, Object>> bundlesInQueue) t
...
 		final PublishAuditHistory localHistory = bundleAudit.getStatusPojo();
 		final String auditedBundleId = bundleAudit.getBundleId();
+		//Info need to generate Growl Notification
+		final Bundle bundle = APILocator.getBundleAPI().getBundleById(auditedBundleId);
+		final boolean isBundleNameGenerated = bundle.getName().startsWith("bundle-");
+		String notificationMessage = "";
+		String notificationMessageArgument = "";
+		long notificationLife = 5000;
+		MessageSeverity notificationSeverity = MessageSeverity.SUCCESS;
+
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:318]: All these arguments could become an object called `NotificationParams` and use a builder pattern. 
E.g. 
```new NotificationParamsBuilder().withMessage("").withMessageArgument("").withNotificationLife("").withSeverity(SUCCESS)```

and then call the .build() when passing it to the private method `sendNotification(builder.build())`

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java (hunk 3 above)
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:413]: not sure if this should be configurable in order to get more detail if wanted

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java (hunk 6 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:321]: using builder

File: dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java (hunk 3 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java:415]: This is to keep consistency with the Status/History Tab

**Review by dsilvam — APPROVED** [review]

**Pull request merged**